keps/sig-network/4559-kubelet-probes/README.md

```diff
@@ -0,0 +1,1037 @@
+   rather than being sent over the pod network (unless you use the
+   deprecated `Host` field to request an alternate probe endpoint; we
+   probably shouldn't just move those kinds of probes into the pod
+   network namespace since that might require a "hairpin" connection
+   to work when it wasn't required before).
+
+2. Leave the existing probe types with their existing semantics, but
+   deprecate them and add new probe types with the
+   inside-the-network-namespace semantics. (`v1.ProbeHandler`
+   currently has `httpGet` and `tcpSocket` fields; these could
+   potentially become just `http` and `tcp`. It's less obvious what
+   `grpc` could be renamed to.)
+
+2½. Deprecate-and-replace `httpGet` and `tcpSocket` as in (2), but
+    change the semantics of `grpc` as in (1), since GRPC probes are
+    already more constrained than the TCP and HTTP probes anyway (and
+    I don't have a good suggestion for renaming them).
+
+### Kubelet/Runtime Changes to Fix the NetworkPolicy Hole
+
+Again, a few possible plans. Some of these ideas are pretty bad, but
+may have good ideas within them that could be remixed into other
+plans...
+
+1. Rather than performing pod probes from the node IP, each kubelet
+   could launch a "probing pod", and perform probes from there. A
+   created-by-default AdminNetworkPolicy could force all pods to
+   accept connections from the kubelet probe pods. This is perhaps
+   _slightly_ better than the current situation since allowing
+   connections from the probe pods would not allow any unintended
+   additional traffic from anyone else the way that allowing
+   connections from the node IP does.
+
+     - The simplest / most portable way to do this would be to have
+       the "probing pod" be an ordinary pod-network pod, though (given
+       current AdminNetworkPolicy rules) this would means that all
+       pods on all nodes would have to accept connections from all
+       probing pods, meaning an attacker who was able to compromise
+       any probing pod would then be able to connect to any pod in the
+       cluster.
+
+     - Alternatively, the "probing pod" could be a static pod with a
+       special node-local IP, allocated independently of the normal
+       pod network. (Handwave handwave handwave...)
+
+2. Kubelet could perform probes inside pods without needing any
+   changes to the runtime, by attaching an ephemeral container to each
+   pod, containing some `kubelet-probe` binary that would run the
+   probes and report the results back to kubelet (by some means). Most
+   plausibly, this would be a single container that would persist
+   through the life of the pod, periodically running probes, rather
+   than being a new ephemeral container attached each time kubelet
+   needed to perform a probe. Kubelet could perhaps attach such a
+   container to the pod at construction time, via CRI calls, without
+   exposing the container to the kubernetes API, like it does with the
```

Comment:
there can be a mismatch on resources consumption also, and the whole lifecycle of the pod will be compromised, if something happens with the probe container do we block the pod? how can user debug something if is hidden? it has a lot of implications we need to consider carefully